### PR TITLE
Eliminate reserves bad debt allocation and add margin to TP 

### PR DESCRIPTION
**Pull request by prat-gpt**
## Description

* Add 1.04 factor in borrower collateralization.
* Update `npTpRatio` to '1 + sqrt(r)/2'.
* Remove 0.995 factor from claimable reserves calculation.
* Update bond factor calculation to minimum 0.005.
* Allow only up to half of the origination fee to be used to settle bad debt till 144 hours of kicktime.
## Purpose

* Solves the issue discovered in Sherlock audit -> https://github.com/sherlock-audit/2023-09-ajna-judging/issues/71
* Solves what we internally have called `Dmitrii - 1`, an issue discovered by @dmitriia  (although not eligible for the Sherlock report) ->
```
There will be more reserves as borrowers can be heavily penalized in auctions happening during up swings due to takePenalty introduction.
If pool rate is high, bondFactor is maximized to 3%, but auction ended at NP due to an upswing. Kicker gets nothing, borrower is penalized with 3.75% of the settled debt. This happens over and over again, reserves are pumped, which is good for token price. This may be an unfair mechanic for borrowers?
```

## Tasks

- [ ] Changes to protocol contracts are covered by unit tests executed by CI.
- [x] Protocol contract size limits have not been exceeded.
- [ ] Gas consumption for impacted transactions have been compared with the target branch, and nontrivial changes cited in the _Impact_ section above.
- [ ] Scope labels have been assigned as appropriate.
- [ ] Invariant tests have been manually executed as appropriate for the nature of the change.

### CODE REVIEW

File: src/libraries/external/SettlerActions.sol [hunk 1]
```diff
@@ -22,7 +22,7 @@ import {
 
 import {
     _auctionPrice,
-    _indexOf,
+   _indexOf,
```
Comment (EdNoepel) [inline src/libraries/external/SettlerActions.sol]: nit: alignment

File: src/libraries/helpers/PoolHelper.sol [hunk 2]
```diff
@@ -165,7 +165,7 @@ import { Maths }   from '../internal/Maths.sol';
         else {
             //slither-disable-next-line divide-before-multiply
             collateral_ = (collateral_ / Maths.WAD) * Maths.WAD; // use collateral floor
-            return Maths.wmul(collateral_, price_) >= debt_;
+            return  Maths.wmul(collateral_, price_) >= Maths.wmul(1.04 * 1e18, debt_);
```
Comment (EdNoepel) [inline src/libraries/helpers/PoolHelper.sol]: nit: extra space after `return`

**Review by EdNoepel — COMMENTED** [review]
Looks good.  Will need to change target to merge branch once it's created.

**Review by mattcushman — APPROVED** [review]
lgtm

File: src/libraries/external/SettlerActions.sol (hunk 1 above)
Comment (prat-gpt) [reply src/libraries/external/SettlerActions.sol]: Fixed [148f0d8](https://github.com/ajna-finance/contracts/pull/962/commits/148f0d8d0eb47b1558361cb03b260487ecb0205f)

File: src/libraries/helpers/PoolHelper.sol (hunk 2 above)
Comment (prat-gpt) [reply src/libraries/helpers/PoolHelper.sol]: Fixed [148f0d8](https://github.com/ajna-finance/contracts/pull/962/commits/148f0d8d0eb47b1558361cb03b260487ecb0205f)

**Review by EdNoepel — COMMENTED** [review]
This seems ready for merge branch.

**Review by ith-harvey — COMMENTED** [review]
src changes look good! ready to work on unit tests.

**Review by EdNoepel — APPROVED** [review]

**Review by ith-harvey — APPROVED** [review]
LGTM

File: src/libraries/external/SettlerActions.sol [hunk 3]
```diff
@@ -141,8 +141,16 @@ library SettlerActions {
                 reserveAuction_.unclaimed;
 
             // settle debt from reserves (assets - liabilities) if reserves positive, round reserves down however
+            // capped at half of the origination fee rate, based on current book fees
             if (assets > liabilities) {
-                borrower.t0Debt -= Maths.min(borrower.t0Debt, Maths.floorWdiv(assets - liabilities, poolState_.inflator));
+                uint256 t0ReserveSettleAmount = Maths.min(Maths.floorWdiv(assets - liabilities, poolState_.inflator), borrower.t0Debt);
+
+                // if auction has not expired, settle up to the borrower reserve limit
+                if((block.timestamp - kickTime < 144 hours) && (Deposits.treeSum(deposits_) > 0)) {
```
Comment (MikeHathaway) [inline src/libraries/external/SettlerActions.sol:149]: Could be good to expand comment to explain what the 144 hours represents here

**Review by MikeHathaway — APPROVED** [review]
LGTM

File: src/libraries/helpers/PoolHelper.sol [hunk 4]
```diff
@@ -198,7 +198,8 @@ import { Maths }   from '../internal/Maths.sol';
         // Use collateral floor for NFT pools
         if (type_ == uint8(PoolType.ERC721)) {
             //slither-disable-next-line divide-before-multiply
-            collateral_ = (collateral_ / Maths.WAD) * Maths.WAD;
+            collateral_ = (collateral_ / Maths.WAD) * Maths.WAD; // use collateral floor
+            return Maths.wmul(collateral_, price_) >= Maths.wmul(1.04 * 1e18, debt_);
         }
         
         return Maths.wmul(collateral_, price_) >= debt_;
```
Comment (MikeHathaway) [inline src/libraries/helpers/PoolHelper.sol]: Do we also need to multiply by 1.04 here?

Comment (prat-gpt) [reply src/libraries/helpers/PoolHelper.sol]: Yes, missed it while merging `develop` I think. Fixed [a5939b9](https://github.com/ajna-finance/contracts/pull/962/commits/a5939b9820ef43e460becb669a8f3c738aa6bb5a)

File: src/libraries/external/SettlerActions.sol (hunk 3 above)
Comment (prat-gpt) [reply src/libraries/external/SettlerActions.sol:149]: Added [a5939b9](https://github.com/ajna-finance/contracts/pull/962/commits/a5939b9820ef43e460becb669a8f3c738aa6bb5a)

**Pull request merged**